cmd/shim/secret_scrub_test.go

```diff
@@ -133,24 +146,62 @@ func TestLoadSecretsToScrubFromFiles(t *testing.T) {
 	})
 }
 
-var (
-	//nolint:typecheck
-	//go:embed testdata/id_ed25519
-	sshSecretKey string
+func TestScrubSecretWrite(t *testing.T) {
+	t.Parallel()
+	envMap := map[string]string{
+		"secret1":      "secret1 value",
+		"secret2":      "secret2",
+		"sshSecretKey": sshSecretKey,
+		"sshPublicKey": sshPublicKey,
+	}
 
-	//nolint:typecheck
-	//go:embed testdata/id_ed25519.pub
-	sshPublicKey string
-)
+	env := []string{}
+	for k, v := range envMap {
+		env = append(env, fmt.Sprintf("%s=%s", k, v))
+	}
 
-func TestScrubSecretWrite(t *testing.T) {
-	secrets := []string{
+	envNames := []string{
 		"secret1",
-		"secret with space ",
-		sshSecretKey,
-		sshPublicKey,
+		"secret2",
+		"sshSecretKey",
+		"sshPublicKey",
 	}
-	s := "Not secret\nsecret1\nsecret with space\n" + sshSecretKey + "\n" + sshPublicKey
-	got := scrubSecretBytes(secrets, []byte(s))
-	require.Equal(t, "Not secret\n***\n***\n***\n***\n***\n***\n***\n***\n***\n\n***\n", string(got))
+
+	out := new(bytes.Buffer)
+	w, err := NewSecretScrubWriter(out, "/", fstest.MapFS{}, env, core.SecretToScrubInfo{
+		Envs:  envNames,
+		Files: []string{},
+	})
+	require.NoError(t, err)
+
+	t.Run("multiline secret", func(t *testing.T) {
+		input := "aaa\n" + sshSecretKey + "\nbbb\nccc"
+		_, err := w.Write([]byte(input))
+		require.NoError(t, err)
+		require.Equal(t, "aaa\n***\n***\n***\n***\n***\n***\n***\n\nbbb\nccc", out.String())
+		out.Reset()
+
+		_, err = w.Write([]byte(sshSecretKey))
+		require.NoError(t, err)
+		require.Equal(t, "***\n***\n***\n***\n***\n***\n***\n", out.String())
```

Comment:
I think it's debatable whether this actually matters or not, but I wonder if it would be slightly better to replace multiline secrets with a single `***` rather than 1 `***` per-line. *Technically* we are leaking the number of lines in the secret here, which could plausibly matter for some limited scenarios.

If it's really easy to switch to replacing the whole thing with a single `***`, I'd vote we just do that, but if that throws a huge wrench into stuff then we can discuss more about whether that actually matters.

Reply:
@sipsma Thanks for the review! This was my attempt to cover functionality of the existing PR, I agree with you.
Agree with replacing the whole thing with a single `***`, will spend some time on it today.

Reply:
@sipsma I've reworked this a bit, I'm still investigating the `TestBigWhitespaceSecretScrubbed` integration test